rts/at/annotation/README.md

```diff
@@ -0,0 +1,59 @@
+    
+    curl -XPOST http://localhost:8082/trip/book?hotelName=Rex
+
+Since the `bookTrip` resource is annotated with `@SRA(value = SRA.Type.REQUIRED, end = false)`, the curl request will start a new REST-AT transaction and you can verify that it is running by querying the coordinator:  `curl http://localhost:8080/rest-at-coordinator/tx/transaction-manager`.
+
+EXPECTED OUTPUT
+---------------
+    SRA: 0_ffff0a4cf0b6_-568c278f_6458bef8_7fb: Updating hotel participant state to: TransactionPrepared
+    SRA: 0_ffff0a4cf0b6_-568c278f_6458bef8_7fb: Updating trip participant state to: TransactionPrepared
+    SRA: 0_ffff0a4cf0b6_-568c278f_6458bef8_7fb: Updating trip participant state to: TransactionCommitted
+    SRA: 0_ffff0a4cf0b6_-568c278f_6458bef8_7fb: Updating hotel participant state to: TransactionCommitted
+    {"cancelPending":false,"details":[{"cancelPending":false,"details":[],"encodedId":"","id":"","name":"","quantity":0,"status":"CONFIRMED","type":""}],"encodedId":"0_ffff0a4cf0b6_-568c278f_6458bef8_7fb","id":"0_ffff0a4cf0b6_-568c278f_6458bef8_7fb","name":"Aggregate Booking","quantity":1,"status":"PROVISIONAL","type":"Trip"}
+
+Example of how to manually test that the coordinator is working using curl
+
+    curl -H "Content-Type: application/x-www-form-urlencoded" -X POST http://localhost:8080/rest-at-coordinator/tx/transaction-manager
+    curl http://localhost:8080/rest-at-coordinator/tx/transaction-manager
+    curl -X PUT --data txstatus=TransactionCommitted http://localhost:8080/rest-at-coordinator/tx/transaction-manag er/0_ffff0a4cf0b6_-568c278f_6458bef8_7fb/terminator
+    curl http://localhost:8080/rest-at-coordinator/tx/transaction-manager
+
+WHAT JUST HAPPENED?
+-------------------
+
+1. We started REST-AT coordinator which is available to use in wildfly.
+
+2. We verified REST-AT coordinator is running by invoking the transaction manager url.
+
+3. Build and run the REST-AT annotation application individually as shown, it starts all three microservice on different port i.e 8082, 8083 and 8084 which is configured in application.properties such as in [flight-service](flight-service/src/main/resources/application.properties), [hotel-service](hotel-service/src/main/resources/application.properties) and [trip-service](flight-service/src/main/resources/application.properties).
+
+4. Invoke book trip which start REST-AT transactions and calls individual hotel and flight service.
+
+5. The example performed an HTTP POST request, a two-phase aware participant is registered with the coordinator by POSTing on the enlisting resource. A successful POST will return HTTP code 200 (ok).
+
+6. If the HTTP status code returned is 202 in  then [/prepare/{id}](https://github.com/jbosstm/narayana/blob/0e45bcb6d5277131801aa7343fcee81068f66d0f/rts/at/tx/src/main/java/org/jboss/jbossts/star/client/SRAParticipant.java#L98) is being called and update the status to TransactionPrepared along with id.
```

Comment:
202 is the `default` [branch of the switch statement](https://github.com/jbosstm/narayana/blob/0e45bcb6d5277131801aa7343fcee81068f66d0f/rts/at/tx/src/main/java/org/jboss/jbossts/star/client/SRAParticipant.java#L166) whereas it is the `TransactionPrepared` branch that  returns 200.

Also can you explain what `along with id` means, your sentence includes the cryptic text `[/prepare/{id}]` which I think you intend to correspond to a URI path template for the `@Path` annotation on the `SRAParticipant` JAX-RS helper methods, if so the id represents the transaction id. But these JAX-RS methods do not return the transaction id (if you look at the SRAParticipant.java source file you should understand my reasoning).

That said, I think steps 6, 7 and 8 are referring to the responses returned by the participants back to the filter [that implements the REST-AT protocol] running in the TripService - but those calls are not visible to the developer, so if you are going to mention them at all then you would need to explain how it works (ie explain what developer responsibilities, with respect to the REST-AT protocol, are being fulfilled by the SRAParticipant helper class).

Reply:
Hi @mmusgrov Could you please provide me what line needs to be added here exactly?

Reply:
I think you should be filling out this section of the README. That said, I did give a hint in the last paragraph of my comment.

Reply:
Hi @mmusgrov I've updated the changes, could you please check and resolve the conversation?
Thanks.